src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -169,7 +174,9 @@ def _crawl(self) -> list[Grant]:
             tasks.append(partial(fn, table=table.name))
         catalog_grants, errors = Threads.gather(f"listing grants for {catalog}", tasks)
         if len(errors) > 0:
-            # TODO: https://github.com/databrickslabs/ucx/issues/406
+            for _e in errors:
+                self._failure_reporter.report(ObjectFailure.make(_e))
+            self._failure_reporter.flush()
```

Comment:
Method flush() will be forgotten to get called 
```suggestion
            self._failure_reporter.record(errors)
```